pilot/cmd/pilot-agent/main.go

```diff
@@ -136,9 +136,10 @@ var (
 				role.IPAddresses = append(role.IPAddresses, ipAddr...)
 			}
 
-			// No IP addresses provided, append 127.0.0.1
+			// No IP addresses provided, append 127.0.0.1 for ipv4 and ::1 for ipv6
 			if len(role.IPAddresses) == 0 {
 				role.IPAddresses = append(role.IPAddresses, "127.0.0.1")
+				role.IPAddresses = append(role.IPAddresses, "::1")
```

Comment:
Did you test on an IP4-only machine (with ip6 stack disabled ) ? I doubt it'll work. Maybe check if the list of local addresses has a FE80 address.